packages/Confirmation/src/components/TriggerButton/TriggerButton.js

```diff
@@ -1,40 +1,54 @@
 import React from "react";
 import PropTypes from "prop-types";
 import Button from "@paprika/button";
+import RawButton from "@paprika/raw-button";
+
+export const ButtonTypes = {
+  ICON: "icon",
+  RAW: "raw",
+  SIMPLE: "simple",
+};
+
+const ButtonComponentMap = {
+  icon: Button.Icon,
+  raw: RawButton,
+  simple: Button,
+};
 
 const propTypes = {
   children: PropTypes.node,
   confirmId: PropTypes.string,
-  icon: PropTypes.node,
-  isConfirmOpen: PropTypes.bool.isRequired,
-  onOpenConfirm: PropTypes.func.isRequired,
+  buttonType: PropTypes.oneOf(ButtonTypes.ALL),
```

Comment:
You are missing this line:
`ButtonTypes.ALL = Object.values(ButtonTypes);`

Which is causing a console warning.